composer/utils/checkpoint.py

```diff
@@ -484,6 +484,20 @@ def _restore_checkpoint(
             exclude_algorithms=exclude_algorithms,
             algorithm_passes=algorithm_passes,
         )
+        step_to_resume_from = state.timestamp.batch.value
+        avg_step_to_resume_from = state.device.tensor_to_device(
```

Comment:
Any particular reason we're doing this with floats? the timestamp should be an int and then we don't have any risk of floating point error messing things up

Reply:
oh I mainly did it to avoid rounding error when averaging.

Reply:
ok changed to min-max comparison